app/models/manageiq/providers/kubernetes/container_manager/metrics_capture.rb

```diff
@@ -37,8 +37,19 @@ class TargetValidationError < RuntimeError; end
       }
     }
 
+    def fetch_target_authentication
+      ems = target.ext_management_system
+      return unless ems
+
+      auth = ems.authentication_type(:metrics_collection)
+      ems.authentications << Authentication.new(:authtype => :metrics_collection) unless auth
```

Comment:
Adding an authentication here is not a nice thing to do.

**We can also:**
a. Create the authentication wile creating a new provider.
b. Add new fields to the provider class.

**Pros:**
1. We support providers created without this authentication - no need to edit the provider and add this authentication to all the previously created providers. 
2. We do not need new fields in the ems class - no need for migrations.
**Cons:**
1. Creating an authentication while reading data - bad thing to do.


@moolitayer @alongoldboim @simon3z help please ?